platform/macos/macos.cmake

```diff
@@ -20,18 +23,46 @@ if(MBGL_WITH_OPENGL)
         mbgl-core
         PRIVATE
             ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/gl/headless_backend.cpp
-            ${PROJECT_SOURCE_DIR}/platform/darwin/src/gl_functions.cpp ${PROJECT_SOURCE_DIR}/platform/darwin/src/headless_backend_cgl.mm
+            ${PROJECT_SOURCE_DIR}/platform/darwin/src/gl_functions.cpp
+            ${PROJECT_SOURCE_DIR}/platform/darwin/src/headless_backend_cgl.mm
     )
     target_link_libraries(
         mbgl-core
         PRIVATE OpenGL::GL
     )
 endif()
 
+add_library(mbgl-platform-loop STATIC)
+
 target_sources(
-    mbgl-core
+    mbgl-platform-loop
```

Comment:
Why do you need to split this for other platforms than node? Is it because node uses macOS platform when on this OS?